packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,156 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test updating the context should update the alias']() {
+    this.render(`{{#with cond1 as |cond|}}{{cond.greeting}}{{else}}False{{/with}}`, {
+      cond1: { greeting: 'Hello' }
+    });
+
+    this.assertText('Hello');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hello');
+
+    this.runTask(() => set(this.context, 'cond1', { greeting: 'Hola'}));
+
+    this.assertText('Hola');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hola');
+
+  }
+
+  ['@test the scoped variable is not available outside the {{with}} block.']() {
+    this.render(`{{name}}-{{#with other as |name|}}{{name}}{{/with}}-{{name}}`, {
+      name: 'Stef',
+      other: 'Yehuda'
+    });
+
+    this.assertText('Stef-Yehuda-Stef');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Stef-Yehuda-Stef');
```

Comment:
In this case, splitting (3) and (4) probably makes sense – you could imagine a bug in the scoping code where changing one would accidentally overwrite the other.